x/mongo/driver/topology/topology.go

```diff
@@ -543,6 +543,24 @@ func (t *Topology) pollSRVRecords() {
 			t.pollHeartbeatTime.Store(false)
 		}
 
+		// If t.cfg.srvMaxHosts is non-zero, randomly select up to SRVMaxHosts hosts from
+		// parsedHosts using the modern Fisher-Yates algorithm.
+		if t.cfg.srvMaxHosts != 0 {
```

Comment:
Same comment here - we can skip this step if the max is greater than the number of hosts we parsed.